src/expiring_hash_map/mod.rs

```diff
@@ -1,5 +1,6 @@
 use futures::ready;
 use futures::stream::Stream;
+use futures::task::AtomicWaker;
```

Comment:
We should try to avoid sync/task types from the futures crate for tokio based ones since those are loom checked.

Reply:
This is a really valuable advantage of the `tokio` types indeed. I'll switch to `Notify`.

Reply:
We changed the API to get rid of the need to have an internal `AsyncWaker`/`Notify`.